<launch>
  <arg name="main" default="true" />
  <arg name="json_dir" />
  <arg name="rviz" default="true" />

  <!-- set bin_contents rosparam -->
  <node name="set_bin_contents"
        pkg="jsk_arc2017_common" type="set_location_in_rosparam.py">
    <remap from="~param" to="/bin_contents" />
    <rosparam subst_value="true">
      location: bins
      json_dir: $(arg json_dir)
    </rosparam>
  </node>

  <!-- work order data -->
  <node pkg="jsk_arc2017_common" type="work_order_publisher.py" name="strategic_work_order" respawn="true">
    <rosparam subst_value="true">
      json_dir: $(arg json_dir)
    </rosparam>
  </node>

  <!-- parameter -->
  <param name="/left_hand/state" value="" />
  <param name="/left_hand/target_bin" value="" />
  <param name="/right_hand/state" value="" />
  <param name="/right_hand/target_bin" value="" />

  <node name="rqt_yn_btn"
        pkg="jsk_rqt_plugins" type="rqt_yn_btn">
    <remap from="~" to="/rviz/yes_no_button" />
  </node>
  <group if="$(arg rviz)">
    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find jsk_2016_01_baxter_apc)/rvizconfig/pick.rviz">
    </node>
  </group>

  <!-- robot main -->
  <node if="$(arg main)" pkg="jsk_arc2017_baxter" type="pick-main.l" name="left_hand_main"
        args="&quot;(progn (pick-init :larm-controller) (pick-mainloop :larm))&quot;"
        respawn="true" output="screen">
    <env name="DISPLAY" value="" />
  </node>

  <node if="$(arg main)" pkg="jsk_arc2017_baxter" type="pick-main.l" name="right_hand_main"
    args="&quot;(progn (pick-init :rarm-controller) (pick-mainloop :rarm))&quot;"
    respawn="true" output="screen">
    <env name="DISPLAY" value="" />
  </node>

  <!-- visualization -->
  <!-- TODO: json visualization -->
  <!--
  <node name="visualize_json"
    pkg="jsk_apc2016_common" type="visualize_pick_json.py"
    args="$(arg json)" />
  -->
    <node name="rosconsole_overlay_text"
        pkg="jsk_rviz_plugins" type="rosconsole_overlay_text.py">
    <rosparam>
      nodes:
        - /left_hand
        - /right_hand
    </rosparam>
  </node>

  <node name="team_k_desktop_xdisplay"
        pkg="jsk_baxter_startup" type="xdisplay_image_topic.py"
        args="/visualize_json/output">
  </node>

</launch>
